### ISSUE TITLE
list not supported of separate fields

### ISSUE THREAD

**Issue by holtkamp** [dragonmantank/cron-expression#124]
When using the public functions of separate Cron fields (like MinutesField, HoursField, DayOfWeekField, etc), it seems "lists" are not respected / supported 

For example:

```php
$weekdayField = new DayOfWeekField();
var_dump($weekdayField->isSatisfied(5, '5,6')); //Returns TRUE
var_dump($weekdayField->isSatisfied(6, '5,6')); //Returns FALSE while TRUE is expected
```

It seems that support for lists is implemented as part of [`CronExpression::getRunDate()`](https://github.com/dragonmantank/cron-expression/blob/76ec7333bd29db2fe3c883b311412a12e5022796/src/Cron/CronExpression.php#L406-L417) which is part of using `CronExpression::isDue()` and thereby `CronExpression::getNextRunDate()`.

Maybe this is on purpose, it might be good to document then? And maybe functions like "isSatisfied" and "isSatisfiedBy" need to be marked as `@internal`?

Or even better: add support for lists at "field" level...

**Comment by dragonmantank** [comment]
Yeah, I think ultimately this would be better marked as `@internal`. There be all sorts of weird stuff going on in there (which is actually an impetus for the upcoming v4 rewrite, there's a lot of weird code that isn't shared across various checks). I'll update `master` here in a moment, and close this once I push out an official release for it.

I would figure out a way to make it work, but a ton of code is getting refactored and the whole validate/should-I-run stuff is rewritten.

**Comment by dragonmantank** [comment]
This should be fixed with v3.2.2
